builder/swig/java.bzl

```diff
@@ -0,0 +1,194 @@
+        ),
+        "package": attr.string(
+            doc = "Java package for which to generate the sources",
+            mandatory = True,
+        ),
+        "_jni_header": attr.label(
+            default = Label("@bazel_tools//tools/jdk:jni_header"),
+            allow_single_file = True,
+        ),
+        "jni_md_header": attr.label(
+            allow_single_file = True,
+        ),
+        "_swig": attr.label(
+            default = Label("@swig//:swig"),
+            allow_single_file = True,
+            executable = True,
+            cfg = "host",
+        ),
+        "_swig_interface_template": attr.label(
+            default = Label("//builder/swig:template.i"),
+            allow_single_file = True,
+        ),
+        "_zipper": attr.label(
+            default = Label("@bazel_tools//tools/zip:zipper"),
+            allow_single_file = True,
+            executable = True,
+            cfg = "host",
+        ),
+    },
+)
+
+
+def swig_java_wrapper(**kwargs):
+    # workaround for select() not being allowed as a default argument
+    # cf. https://github.com/bazelbuild/bazel/issues/287
+    _swig_java_wrapper(
+        jni_md_header = select({
+            "@vaticle_dependencies//util/platform:is_mac": Label("@bazel_tools//tools/jdk:jni_md_header-darwin"),
+            "@vaticle_dependencies//util/platform:is_linux": Label("@bazel_tools//tools/jdk:jni_md_header-linux"),
+            "@vaticle_dependencies//util/platform:is_windows": Label("@bazel_tools//tools/jdk:jni_md_header-windows"),
+        }),
+        **kwargs,
+    )
+
+
+def swig_java(name, lib, shared_lib_name=None, **kwargs):
+    swig_wrapper_name = name + "__swig"
+    swig_java_wrapper(
+        name = swig_wrapper_name,
+        class_name = name,
+        lib = lib,
+        **kwargs,
+    )
+
+    if not shared_lib_name:
+        shared_lib_name = "lib" + name
+
+    def swig_cc_binary(shared_lib_filename):
+        # name doesn't accept select() either
+        native.cc_binary(
+            name = shared_lib_filename,
+            deps = [lib, swig_wrapper_name],
+            srcs = [swig_wrapper_name],
+            linkshared = True,
+        )
+
+    select({
+        "@vaticle_dependencies//util/platform:is_mac": swig_cc_binary(shared_lib_name + ".dylib"),
+        "@vaticle_dependencies//util/platform:is_linux": swig_cc_binary(shared_lib_name + ".so"),
+        "@vaticle_dependencies//util/platform:is_windows": swig_cc_binary(shared_lib_name + ".lib"),
+    })
+
+    native.alias(
+        name = shared_lib_name,
+        actual = select({
+            "@vaticle_dependencies//util/platform:is_mac": (shared_lib_name + ".dylib"),
+            "@vaticle_dependencies//util/platform:is_linux": (shared_lib_name + ".so"),
+            "@vaticle_dependencies//util/platform:is_windows": (shared_lib_name + ".lib"),
+        })
+    )
+
+    native.java_library(name = name, srcs = [swig_wrapper_name], deps = [shared_lib_name])
```

Comment:
Heads up to selves: when we distribute the final JAR, we need to make sure the DYLIB/shared library is findable by the java load library command

Reply:
We can either package into this JAR or into the parent JAR